controllers/generic_network_controller.go

```diff
@@ -88,17 +88,15 @@ func (r *genericNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 			// Request object not found, could have been deleted after reconcile request.
 			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
 			// Return and don't requeue
-			return reconcile.Result{}, nil
+			return reconcile.Result{}, r.garbageCollect(ctx)
```

Comment:
for network CR instances that are in the operator namespace and point to a different namespace, are we relying on garbageCollect to clean them ? 

i think the approach is problematic, since we are removing finalizers at L97, once the object is deleted, it will be removed from k8s API. if for some reason garbage collect fails and controller is restarted we will never retry.

Reply:
sounds good. The first commit is about adding the owner-ref annotation and ensure 2 NetworkObject are not clashing on the same NetAttachDef. The second commit is about using the `owner-ref` annotation on the netattachdef instead of networkobject's finalizers, which I was a sort of bonus change.

I'm dropping the second commit to focus this PR on having the `owner-ref` annotation. We can discuss about dropping the finalizers later, if needed